Vostok.Applications.Scheduled/Schedulers/OnDemandScheduler.cs

```diff
@@ -63,4 +63,29 @@ public void OnFailedIteration(IScheduler source, Exception error)
 
         public override string ToString() => "OnDemand";
     }
+
+    internal class OnDemandSchedulerWithArgument<TArg> : OnDemandScheduler
+    {
+        private TArg lastArgument;
+        
+        public void Demand(TArg argument)
+        {
+            lock (Sync)
```

Comment:
Lock under the same object twice.

At first I thought it should be a deadlock, but this actually works.

Nevertheless, it's not a problem to rewrite this using other sync object

Reply:
Not an issue, `Monitor`-based lock are reentrant by design.